app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -344,11 +352,42 @@ public boolean onOptionsItemSelected(MenuItem item)
                 }
                 rotationEnabled = !rotationEnabled;
                 return true;
+
+            case R.id.action_star_unstar:
+                if (starred)
+                {
+                    setStarinDB(item, loyaltyCardId, false);
+                }
+                else
+                {
+                    setStarinDB(item, loyaltyCardId, true);
+                }
+                starred = !starred;
+                return true;
```

Comment:
This whole thing can probably be simplified as:

```
starred = !starred;
setStarInDB(item, loyaltyCardId, starred);
return true;
```

Reply:
Correct! It was just copied from similar case R.id.action_lock_unlock above ....
I changed the code